rrule/src/parser/regex.rs

```diff
@@ -50,7 +44,15 @@ impl ParsedDateString {
     /// Parses a date string with format `YYYYMMDD(THHMMSSZ)` where the part in parentheses
     /// is optional. It returns [`ParsedDateString`].
     pub(crate) fn from_ical_datetime(val: &str) -> Result<Self, ParseError> {
+        static DATESTR_RE: OnceLock<Regex> = OnceLock::new();
```

Comment:
Should this be a global variable instead to avoid the regex being compiled each time this function is called?

Reply:
Isn't that the effect of the `static` keyword in this context?

Reply:
From https://doc.rust-lang.org/reference/items/static-items.html:

> A static item is similar to a [constant](https://doc.rust-lang.org/reference/items/constant-items.html), except that it represents a precise memory location in the program. All references to the static refer to the same memory location.

Reply:
Indeed, as @WhyNotHugo pointed out, declaring a `static` within a function body as opposed to directly within a module only has the effect of limiting its visibility scope. All other semantics, including having a single instance of the variable in memory, stay the same.

Reply:
Right, I suspected that was the case. Thanks for confirming!